packages/ui5-config/src/ui5config.ts

```diff
@@ -335,19 +383,113 @@ export class UI5Config {
     /**
      * Adds a instance of the mockserver middleware to the config.
      *
-     * @param path option path that is to be mocked
-     * @param annotationsConfig optional annotations config that is to be mocked
+     * @param dataSourcesConfig - annotations config that is to be mocked
+     * @param annotationsConfig - annotations config that is to be mocked
+     * @param appRoot - root to the application
      * @returns {UI5Config} the UI5Config instance
      * @memberof UI5Config
      */
-    public addMockServerMiddleware(path?: string, annotationsConfig?: MockserverConfig['annotations']): this {
+    public addMockServerMiddleware(
+        dataSourcesConfig: DataSourceConfig[],
+        annotationsConfig: MockserverConfig['annotations'],
+        appRoot?: string
+    ): this {
         this.document.appendTo({
             path: 'server.customMiddleware',
-            value: getMockServerMiddlewareConfig(path, annotationsConfig)
+            value: getMockServerMiddlewareConfig(dataSourcesConfig, annotationsConfig, appRoot)
         });
         return this;
     }
 
+    /**
+     * Adds a service configuration to an existing sap-fe-mockserver middleware keeping any existing service configurations. If the config does not contain a sap-fe-mockserver middleware, an error is thrown.
+     *
+     * @param dataSourceConfig - dataSource config from manifest to add to mockserver middleware services list
+     * @param appRoot - root to the application
+     * @param annotationsConfig - optional, annotations config that is to be mocked
+     * @returns {UI5Config} the UI5Config instance
+     * @memberof UI5Config
+     */
+    public addServiceToMockserverMiddleware(
+        dataSourceConfig: DataSourceConfig,
+        appRoot?: string,
+        annotationsConfig: MockserverConfig['annotations'] = []
+    ): this {
+        const mockserverMiddleware = this.findCustomMiddleware<MockserverConfig>('sap-fe-mockserver');
+        if (!mockserverMiddleware) {
+            throw new Error('Could not find sap-fe-mockserver');
+        } else {
+            // Else append new data to current middleware config and then run middleware update
+            const serviceRoot = `${appRoot ?? './webapp'}/localService/${dataSourceConfig.serviceName}`;
```

Comment:
Let's ignore this for now because that will hopefully be solved with https://github.com/SAP/open-ux-tools/issues/2531 because ui5-config actually should not have dependencies to middlewares (only the other way around).